src/arithmetic.jl

```diff
@@ -0,0 +1,501 @@
+# This file is part of the TaylorSeries.jl Julia package, MIT license
+#
+# Luis Benet & David P. Sanders
+# UNAM
+#
+# MIT Expat license
+#
+
+# Arithmetic operations: +, -, *, /
+
+## Equality ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
+        function =={T<:Number}(a::$T{T}, b::$T{T})
+            a, b = fixorder(a, b)
```

Comment:
But you are creating a whole new `Taylor1` object just to check equality, where you could just be checking that the coeffs that match are equal, and the extra coeffs in the longer vector are all 0.